Remove setting eval mode in observed custom LSTM

Ninja91 opened the pull request:
Summary: Same as title.

Test Plan: CI

Differential Revision: D46831286

HDCharles reviewed (changes requested):
can you fix the failures

Ninja91 committed:
Remove setting eval mode in observed custom LSTM (#103903)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/103903

Same as title.

Test Plan: CI

Differential Revision: D46831286

fbshipit-source-id: ba6803a414381c45cd6b2dde41e14f44d3893e87
